api/schemas/catalog.json

```diff
@@ -94,8 +94,88 @@
             "items": {
                 "type": "string"
             }
+        },
+        "metadata": {
+            "description": "A mapping of non-recursive File resources to their required metadata",
+            "type": "object",
+            "patternProperties": {
+                "^.+$": {
+                    "$ref": "#/definitions/file_metadata"
+                }
+            }
+        },
+        "recursive_metadata": {
+            "description": "A mapping of recursive File resources to a hash of sources and arrays of associated metadata for that resource",
+            "type": "object",
+            "patternProperties": {
+                "^.+$": {
+                    "type": "object",
+                    "patternProperties": {
+                        "^.+$": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/file_metadata"
```

Comment:
I think if we wrap the current `file_metadata.json`schema in a `definitions`, something like:

```json
{
  "definitions": {
    "file_metadata": {
    "title":       "File Metadata",
    "description": "Metadata about a file, directory, or symbolic link",
    "type":        "object",
    ...
    }
  }
}
```

Then I think you can reference the schema from a different file like:

```json
   { "$ref": "file_metadata.json#/definitions/file_metadata" }
```